internal/pkg/utils/utils.go

```diff
@@ -45,3 +46,44 @@ func FieldName(fa *ssa.FieldAddr) string {
 	}
 	return st.Field(fa.Field).Name()
 }
+
+// DecompoeType returns the path, typename, and indicators for if the Type is Named or an Interface
+// Returns empty strings if the type is not *types.Named
+func DecomposeType(t types.Type) (path, name string) {
+	n, ok := t.(*types.Named)
+	if !ok {
+		return
+	}
+
+	if pkg := n.Obj().Pkg(); pkg != nil {
+		path = pkg.Path()
+	}
+
+	return path, n.Obj().Name()
+}
+
+func UnqualifiedName(v *types.Var) string {
+	packageQualifiedName := v.Type().String()
+	dotPos := strings.LastIndexByte(packageQualifiedName, '.')
+	if dotPos == -1 {
+		return packageQualifiedName
+	}
+	return packageQualifiedName[dotPos+1:]
+}
+
+// DecomposeFunction returns the path, receiver, and name strings of a ssa.Function.
+// For functions that have no receiver, returns an empty string for recv.
+// If f is nil, returns empty strings for all return values.
```

Comment:
When would `f` be `nil`?

Reply:
For instance,

```golang
// levee.go:62
				case conf.IsSink(utils.DecomposeFunction(v.Call.StaticCallee())):
```

`StaticCallee()` may be nil for builtins and dynamic-dispatch calls.  See [the `CallCommon` docs](https://godoc.org/golang.org/x/tools/go/ssa#CallCommon) for the particulars.

Reply:
Oh, right! Thanks for explaining.

I feel like handling that is the caller's responsibility. To me, `DecomposeFunction`'s job is to take a `Function` and decompose it into `path`, `recv` and `name`. It shouldn't be called with a `nil` value. The caller shouldn't be trying to decompose a non-existent `StaticCallee()`, so if `StaticCallee()` is `nil`, it shouldn't call `DecomposeFunction` at all. 

I think it would be clearer to have the `nil` check in the caller. When I see `v.Call.StaticCallee()`, I can easily see why that might be `nil`. When I see `if f == nil {`, it's not obvious to me why `f` could be nil, and in fact I don't think it should ever be.